lm_eval/tasks/ja/jaqket_v2.py

```diff
@@ -94,6 +140,105 @@ def doc_to_target(self, doc):
+        assert not provide_description, (
+            "The `provide_description` arg will be removed in future versions. To prepend "
+            "a custom description to the context, supply the corresponding string via the "
+            "`description` arg."
+        )
+        if provide_description is not None:
+            # nudge people to not specify it at all
+            print(
+                "WARNING: provide_description is deprecated and will be removed in a future version in favor of description_dict"
+            )
+
+        if hasattr(self, "FEWSHOT_SEP"):
+            FEWSHOT_SEP = self.FEWSHOT_SEP
+        elif hasattr(self, "SEP"):
+            FEWSHOT_SEP = f"{self.SEP}{self.SEP}"
+        else:        
+            FEWSHOT_SEP = "\n\n"
+            
+        if description:
+            description += FEWSHOT_SEP
+        elif hasattr(self, "DESCRIPTION"):
+            description = self.DESCRIPTION
+        else:
+            description = ""
+
+        if num_fewshot == 0:
+            labeled_examples = ""
+        else:
+            # for sets with no training docs, draw from other set *but ensure no overlap with current doc*
+            if self.has_training_docs():
+                fewshotex = self.fewshot_examples(k=num_fewshot, rnd=rnd)
+            else:
+                if self._fewshot_docs is None:
+                    self._fewshot_docs = list(
+                        self.validation_docs()
+                        if self.has_validation_docs()
+                        else self.test_docs()
+                    )
+
+                fewshotex = rnd.sample(self._fewshot_docs, num_fewshot + 1)
+
+                # get rid of the doc that's the one we're evaluating, if it's in the fewshot
+                fewshotex = [x for x in fewshotex if x != doc][:num_fewshot]
+
+            labeled_examples = (
+                FEWSHOT_SEP.join(
+                    [
+                        self.doc_to_answering_text(doc, self.FALLBACK_DOC) + self.doc_to_target(doc)
+                        for doc in fewshotex
+                    ]
+                )
+                + FEWSHOT_SEP
+            )
+
+        example = self.doc_to_text(doc)
+        return description + labeled_examples + example
+
+    def preprocess_ctx(self, ctx, max_length):
+        # if ctx fits in max length, return
+        if len(self.tokenizer.encode(ctx)) <= max_length:
+            return ctx
+        
+        # if ctx is too long, split on a tag that separates each example
+        description, remainder = ctx.split(self.FEWSHOT_SEP, 1)
+        ctxs = remainder.split(self.FEWSHOT_SEP)
+
+        # if there is no example and still the prompt is too long, fail
+        if len(ctxs) < 2:
+            raise ValueError(f"0-shot description+example doesn't fit in max length. ctx: {ctx}")
+
+        # delete the first example, last is questioning example
+        del ctxs[0]
+
+        # recurse
```

Comment:
```suggestion
        # recur
```

Minor point but "recur" is the proper term.

Reply:
corrected